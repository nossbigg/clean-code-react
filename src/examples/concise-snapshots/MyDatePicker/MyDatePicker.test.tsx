import React from "react";
import { mount, shallow } from "enzyme";
import { MyDatePicker } from "./MyDatePicker";
import { DatePicker } from "antd";

describe("MyDatePicker", () => {
  it("bad: creates an unmaintanably large snapshot", () => {
    const wrapper = mount(<MyDatePicker placeholder="some-placeholder" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("matches snapshot", () => {
    const wrapper = shallow(<MyDatePicker placeholder="some-placeholder" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("passes correct props to DatePicker", () => {
    const wrapper = shallow(<MyDatePicker placeholder="some-placeholder" />);

    const datePicker = wrapper.find(DatePicker);
    expect(datePicker.exists()).toBe(true);

    const expectedProps = { picker: "month", placeholder: "some-placeholder" };
    expect(datePicker.props()).toEqual(expectedProps);
  });
});
