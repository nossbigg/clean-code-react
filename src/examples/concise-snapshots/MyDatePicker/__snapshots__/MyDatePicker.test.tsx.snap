// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyDatePicker bad: creates an unmaintanably large snapshot 1`] = `
<div
  className="ant-picker"
  key="trigger"
  onMouseUp={[Function]}
>
  <div
    className="ant-picker-input"
  >
    <input
      autoComplete="off"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      placeholder="some-placeholder"
      readOnly={true}
      size={12}
      title=""
      value=""
    />
    <span
      className="ant-picker-suffix"
    >
      <span
        aria-label="calendar"
        className="anticon anticon-calendar"
        role="img"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="calendar"
          fill="currentColor"
          focusable="false"
          height="1em"
          key="svg-calendar"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            key="svg-calendar-svg-0"
          />
        </svg>
      </span>
    </span>
  </div>
</div>
`;

exports[`MyDatePicker matches snapshot 1`] = `
<Picker
  picker="month"
  placeholder="some-placeholder"
/>
`;
